### PR TITLE
Added documentation for point in time recovery

### PR DESCRIPTION
**Pull request by arindamnayak**
Added documentation for point in time recovery

### CODE REVIEW

File: content/en/docs/reference/features/recovery.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+title: Point In Time Recovery
+aliases: ['/docs/recovery/pitr','/docs/reference/pitr/']
+---
+
+## Point in time recovery
+
+### Introduction
+
+This enables to recover data in vitess to a past point in time. There can be multiple recovery request active at same time. It is possible to recover across sharding action i.e. you should be able to recover to a time when there were two shards even though at present there are four.
```
Comment (deepthi) [inline content/en/docs/reference/features/recovery.md]:
```suggestion
The Point in Time Recovery feature in Vitess enables recovery of data to a past point in time. There can be multiple recovery requests active at the same time. It is possible to recover across sharding actions, i.e. you can recover to a time when there were two shards even though at present there are four.
```

File: content/en/docs/reference/features/recovery.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+### Usecase(s)
```
Comment (deepthi) [inline content/en/docs/reference/features/recovery.md]:
```suggestion
### Usecases
```

File: content/en/docs/reference/features/recovery.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+- Accidental deletion of data.
+- Corruption of data due to application bugs.
+
+### Precondition
```
Comment (deepthi) [inline content/en/docs/reference/features/recovery.md]:
```suggestion
### Preconditions
```

File: content/en/docs/reference/features/recovery.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+- There should be regular backup taken before point in time.
```
Comment (deepthi) [inline content/en/docs/reference/features/recovery.md]:
```suggestion
- There should be a vitess backup taken before the desired point in time.
```

File: content/en/docs/reference/features/recovery.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+- There should be binlogs available till the required point.
```
Comment (deepthi) [inline content/en/docs/reference/features/recovery.md]:
```suggestion
- There should be continuous binlogs available from the backup time to the desired point in time.
```

File: content/en/docs/reference/features/recovery.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+### Usage
+
+#### Pre-setup
+To use this feature, you need to have regular backup of vitess data and continuous binlogs. Once all this setup is done, you can checkout how to recover.
```
Comment (deepthi) [inline content/en/docs/reference/features/recovery.md]:
```suggestion
To use this feature, you need a usable backup of vitess data and continuous binlogs. .
```

File: content/en/docs/reference/features/recovery.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+Here is how you can take regular backup.
```
Comment (deepthi) [inline content/en/docs/reference/features/recovery.md]:
```suggestion
Here is how you can create a backup.
```

File: content/en/docs/reference/features/recovery.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+`vtctlclient -server localhost:15999 Backup zone1-101`
+
+Here `localhost:15999` is the address of vtctld grpc server. `zone1-101` is the tablet alias of a replica tablet in shard.
+
+To maintain continuous binlog, you need to have a binlog server pointing to the master. You can use [ripple](https://github.com/google/mysql-ripple). You need to take care of following things.
```
Comment (deepthi) [inline content/en/docs/reference/features/recovery.md]:
```suggestion
To maintain continuous binlogs, you need to have a binlog server pointing to the master. You can use [ripple](https://github.com/google/mysql-ripple). You need to take care of the following.
```

File: content/en/docs/reference/features/recovery.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+ - You need to have a highly available binlog server setup, if it is down, we should be able to catch up before master rotates the binlog.
+ - The binlog files should be safely kept at some reliable and recoverable location (i.e. AWS S3, remote file storage). 
+ - In case of any reparenting or master failover, the binlog server should be able to switch to the new master.
+ 
+ Once above is done, you can proceed to send recover request. Here is how you can do a recovery.
```
Comment (deepthi) [inline content/en/docs/reference/features/recovery.md]:
```suggestion
 Once above is done, you can proceed to send a recovery request. Here is how you can do a recovery.
```

**Review by deepthi — COMMENTED** [review]
I have suggested some re-wording of the text. Otherwise looks good.

File: content/en/docs/reference/features/recovery.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+- Accidental deletion of data.
+- Corruption of data due to application bugs.
+
+### Precondition
+- There should be regular backup taken before point in time.
+- There should be binlogs available till the required point.
+
+### Usage
+
+#### Pre-setup
```
Comment (deepthi) [inline content/en/docs/reference/features/recovery.md]: I think this heading can be removed.
```suggestion
```

**Comment by arindamnayak** [comment]
@deepthi , added changes for the PR comments.

**Review by deepthi — APPROVED** [review]
LGTM

**Comment by deepthi** [comment]
@arindamnayak Sorry one more thing that should be fixed.
I failed to notice on the preview that the new page is directly under `Reference`. It should be under `Features`.

**Comment by arindamnayak** [comment]
> @arindamnayak Sorry one more thing that should be fixed.
> I failed to notice on the preview that the new page is directly under `Reference`. It should be under `Features`.

#486 addresses this.

**Pull request merged**